lib/send-batch.ts

```diff
@@ -0,0 +1,81 @@
+import { chunk } from 'lodash'
+import { type Observable, catchError, concatMap, delay, from, map, of, retry, zip } from 'rxjs'
+import { type UserRow, useAppDataStore, useSendBatchState, useTemplateStore } from '~/lib/hooks'
+import { requestGoogleAccessToken } from '~/lib/token'
+import { type SendEmailParams, sendEmailAsync } from './email'
+
+const quotaLimit = 2
+const sendInterval = 1000
+
+type SendEmailStatus = 'success' | 'error'
+
+// Retry send email observable, this never emits error
+const mkSendEmailObservable = (params: SendEmailParams): Observable<SendEmailStatus> =>
+  from(sendEmailAsync(params)).pipe(
+    retry({
+      count: 2,
+      delay: 1000,
+    }),
+    map(() => 'success' as const),
+    catchError(() => of('error' as const))
+  )
```

Comment:
We ahve to be careful about retries.  Retries might overwhelm the 2 per second limit.

Reply:
> We ahve to be careful about retries. Retries might overwhelm the 2 per second limit.

We won't overwhelm the 2 per second limit because we are sending max 2 requests at the same time,  and gmail api call for next set of 2 emails won't happen until the previous ones are completed. (I think rxjs guarantees this)